### PR TITLE
NIFI-12639 Ported all the changes from NIFI-11627 for the support/nifi-1.x branch

### PR DESCRIPTION
**Pull request by dan-s1**
# Summary

[NIFI-12639](https://issues.apache.org/jira/browse/NIFI-12639)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for working this @dan-s1. The automated builds on Java 8 are failing due to some unsupported language features, like `Set.of()`.

**Comment by dan-s1** [comment]
@exceptionfactory Thank you for pointing out the issue. In fact it was the the use of `List.of` for the properties and `Set.of` for the relationships in `ValidateJson` which caused the Java 8 builds to fail.

File: nifi-nar-bundles/nifi-registry-bundle/nifi-registry-service/pom.xml [hunk 1]
```diff
@@ -52,5 +52,20 @@ language governing permissions and limitations under the License. -->
             <groupId>commons-io</groupId>
             <artifactId>commons-io</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.networknt</groupId>
+            <artifactId>json-schema-validator</artifactId>
+            <version>1.0.87</version>
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-registry-bundle/nifi-registry-service/pom.xml:58]: This should be version `1.1.0`.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
@dan-s1 This looks close to completion. I noted one more issue where the version of json-schema-validator should be updated to match the current version of the main branch.

File: nifi-nar-bundles/nifi-registry-bundle/nifi-registry-service/pom.xml (hunk 1 above)
Comment (dan-s1) [reply nifi-nar-bundles/nifi-registry-bundle/nifi-registry-service/pom.xml]: Corrected. Should this dependency be pulled to a `<dependencyManagement>` section in a parent common both to `nifi-nar-bundles/nifi-standard-bundle/pom.xml` and `nifi-nar-bundles/nifi-registry-bundle/nifi-registry-service/pom.xml`so the version could be specified only once?

Comment (exceptionfactory) [reply nifi-nar-bundles/nifi-registry-bundle/nifi-registry-service/pom.xml]: Thanks for the update. This one could go either way. For now, given just two references, it seems alright to leave it as it stands, but if there are any new modules that use it, it makes sense to promote it to a higher level.

**Review by exceptionfactory — APPROVED** [review]
Thanks for backporting the feature and working through the adjustments @dan-s1, the latest version looks good! +1 merging to the support branch.

**Comment by exceptionfactory** [comment]
Merged to support branch in https://github.com/apache/nifi/commit/89ba16a09a6efdef70d60551462d3f4f95195328

**Pull request closed without merging**